### PR TITLE
Add <Viewport />

### PR DESCRIPTION
**Pull request by sandorvasas**
Viewport component that feeds its dimensions into its child grids as if it was the screen resolution.

### CODE REVIEW

File: src/utilities/ScreenProvider/index.jsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/* global window */
+/* eslint no-console: off */
```
Comment (r3nya) [inline src/utilities/ScreenProvider/index.jsx]: Why do you need it here?

File: src/utilities/Viewport/index.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  constructor(props, context) {
+    super(props, context)
+    this.ElementId = `VP${Date.now()}${Math.floor(Math.random()*10000000)}`
+  }
+
+  componentDidMount = () => {
+    this.resizeObserver = new ResizeObserver(entries => {
+      var e = entries && entries[0]
+      if (!e) return;
+      const { top, left, width, height } = e.contentRect
```
Comment (gerbenmeyer) [inline src/utilities/Viewport/index.jsx]: Are `top`, `left`, and `height` ever used?

File: src/utilities/Viewport/index.jsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
+/* global window */
+/* eslint no-console: off */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import throttle from 'lodash/throttle';
+import { getConfiguration, ScreenContext } from '../../config';
+import { getScreenClass as getDefaultScreenClass } from '../../utils';
+import ResizeObserver from 'resize-observer-polyfill'
```
Comment (gerbenmeyer) [inline src/utilities/Viewport/index.jsx]: do we really need an extra dependency if we can just use `window.innerWidth`?

File: src/utilities/ScreenProvider/index.jsx (hunk 1 above)
Comment (gerbenmeyer) [inline src/utilities/ScreenProvider/index.jsx]: What is the purpose of this component?

File: src/grid/Col/index.jsx [hunk 4]
```diff
@@ -4,10 +4,11 @@ import React, { createElement } from 'react';
 import PropTypes from 'prop-types';
 import throttle from 'lodash/throttle';
 import getStyle from './style';
-import { getConfiguration } from '../../config';
+import { getConfiguration, ScreenContextConsumer } from '../../config';
 import { getScreenClass } from '../../utils';
 import { NoGutterContext } from '../Row';
 
+@ScreenContextConsumer
```
Comment (gerbenmeyer) [inline src/grid/Col/index.jsx]: Can't we have a `ScreenClassConsumer`? In the end, only the `rect.width` is actually used, so by providing a screen class directly, we don't have to call `getScreenClass` all the time.

File: src/utilities/Viewport/index.jsx (hunk 3 above)
Comment (sandorvasas) [reply src/utilities/Viewport/index.jsx]: The key point of this component is that we use a parent element's size as if it was window.innerWidth, which provides the ability to subdivide the screen.

File: src/utilities/ScreenProvider/index.jsx (hunk 1 above)
Comment (sandorvasas) [reply src/utilities/ScreenProvider/index.jsx]: I never used eslint and don't intend to in the future. This file was based off of another local source file in the repo.

File: src/utilities/Viewport/index.jsx (hunk 2 above)
Comment (sandorvasas) [reply src/utilities/Viewport/index.jsx]: It's there for inspiration.

**Pull request closed without merging**